packages/@aws-cdk/aws-events-targets/lib/state-machine.ts

```diff
@@ -13,13 +13,29 @@ export interface SfnStateMachineProps {
    * @default the entire EventBridge event
    */
   readonly input?: events.RuleTargetInput;
+
+  /**
+   * Existing IAM role to run the StateMachine Task
+   *
+   * @default A new IAM role is created
+   */
+  readonly role?: iam.IRole;
 }
 
 /**
  * Use a StepFunctions state machine as a target for Amazon EventBridge rules.
  */
 export class SfnStateMachine implements events.IRuleTarget {
+  private readonly role: iam.IRole;
+
   constructor(public readonly machine: sfn.IStateMachine, private readonly props: SfnStateMachineProps = {}) {
+    if (props.role) {
+      this.role = props.role;
+      this.role.grant(new iam.ServicePrincipal('events.amazonaws.com'));
+    } else {
+      this.role = singletonEventRole(machine, []);
```

Comment:
how does this default role work without any policy statements? I'd expect this to have some permissions like `states:StartExecution` and/or whatever APIs it needs to be able to call.

Reply:
The previous policy statement that was used was
```
new iam.PolicyStatement({	      
        actions: ['states:StartExecution'],	
        resources: [this.machine.stateMachineArn],	
      })]
```
We have added this line (L38) to address the above policy since it is more semantically correct:
```
machine.grantStartExecution(this.role);
```

Reply:
got it, it's maybe worth a comment because the function `singletonEventRole` is expecting policy statements. The reason for passing it empty policy statements is not entirely intuitive and could be helpful for contributors in this area in the future. thoughts?